### ISSUE TITLE
Path normalisation/Stripping leading slashes

### ISSUE THREAD

**Issue by ColinHebert** [gotify/server#533]
**Is your feature request related to a problem? Please describe.**
When working with a relatively naive setup of gotify, one may have a trailing slash when specifying the url of the host (`http://localhost/` instead of `http://localhost` for example).
To avoid any issues, the developer using this URL should append `/message` to the configured url to send a message.

This can lead to the fully formed looking like this `http://localhost//mesage`, unfortunately with the way gotify is currently built, the double slash is not normalised and a query at this URL will lead to a 404

**Describe the solution you'd like**
Gotify should normalise the URL it receives to deduplicate slashes when possible, those are not meant to have any bearing on the URL.

**Describe alternatives you've considered**
The alternative is to leave with misconfigurations and hope they are being detected early on by the API user.

**Comment by jmattheis** [comment]
I'm accepting PR's for this.

**Comment by ColinHebert** [comment]
Leaving that for whomever wants to take it.
FWIW, gin, the HTTP server, has a nice parameter `RemoveExtraSlash` which should handle that neatly.
It should be a case of editing this https://github.com/gotify/server/blob/master/router/router.go#L25 to add `g.RemoveExtraSlash = true`.

Caveats, testing is a bit of a chore, while at it, it's worth checking why a new gin engine is created from scratch rather than `gin.Default()` which has a hard crash recovery mechanism and logging enabled (niceties provided out of the box).

**Comment by shubmjagtap** [comment]
hey can i work on this issue ?

**Comment by shubmjagtap** [comment]
@ColinHebert

**Comment by jmattheis** [comment]
@shubmjagtap Yes, but make sure to test this with different reverse proxy configurations, that it'll work correctly.

**Comment by TrungNNg** [comment]
@shubmjagtap  are you still planning to work on this, or would it be okay if I submit the fix? It's been a while so wanted to check first.

@jmattheis 

> make sure to test this with different reverse proxy configurations

could you provide more details on what specific scenarios you'd like tested? and how should I test them - manually?

**Comment by jmattheis** [comment]
Sure, the configs in the gotify docs should continue to work. E.g. https://gotify.net/docs/nginx or https://gotify.net/docs/apache there is no currently no automated way to test this.
